package/appleseed-maya.package.py

```diff
@@ -380,6 +381,339 @@ def run(self, cmdline):
+
+        libs_to_check = set()
+        all_libs = set()
+
+        for bin in glob.glob(os.path.join(os.path.expandvars(self.settings.root_dir), "appleseed", "bin", "*")):
+            libs = self.__get_dependencies_for_file(bin)
+            libs_to_check = libs_to_check.union(libs)
+
+        appleseedpython_libs = self.__get_dependencies_for_file(
+                    os.path.join(self.settings.package_output_path, "scripts", "appleseed", "_appleseedpython.so"))
+        libs_to_check = libs_to_check.union(appleseedpython_libs)
+
+        # find all dependencies
+        while libs_to_check:
+            current_lib = libs_to_check.pop()
+            for lib in self.__get_dependencies_for_file(current_lib):
+                if lib not in all_libs:
+                    libs_to_check.add(lib)
+
+            all_libs.add(current_lib)
+
+        if True:
+            # Print dependencies.
+            info("    Dependencies:")
+            for lib in all_libs:
+                info("      {0}".format(lib))
+
+        # Copy needed libs to lib directory.
+        for lib in all_libs:
+
+            # Don't copy Python, this dependency will be relink to
+            # Maya's Python later in the packaging process
+            if os.path.basename(lib) == "Python":
+                continue
+
+            if True:
+                info("  Copying {0} to {1}...".format(lib, lib_dir))
+            dest = os.path.join(lib_dir, os.path.basename(lib))
+
+            # its possible for the dylib to link ot same lib in multiple paths
+            # check that is not already copied
+            if not os.path.exists(dest):
+                shutil.copy(lib, lib_dir)
+
+    def post_process_package(self):
+        progress("Mac-specific: Post-processing package")
+        self.__fixup_binaries()
+
+    def __fixup_binaries(self):
+        progress("Mac-specific: Fixing up binaries")
+        self.__fix_permissions();
+        self.set_libraries_ids()
+        self.__change_library_paths_in_libraries()
+        self.__change_library_paths_in_executables()
+
+    def __fix_permissions(self):
+        lib_dir = os.path.join(self.settings.package_output_path, "lib")
+        for dirpath, dirnames, filenames in os.walk(lib_dir):
+            for filename in filenames:
+                ext = os.path.splitext(filename)[1]
+                if ext == ".dylib" or ext == ".so":
+                    lib_path = os.path.join(dirpath, filename)
+                    self.run("chmod 777 {}".format(lib_path))
+
+        bin_dir =  os.path.join(self.settings.package_output_path, "bin")
+        for dirpath, dirnames, filenames in os.walk(bin_dir):
+            for filename in filenames:
+                ext = os.path.splitext(filename)[1]
+                if ext != ".py" and ext != ".conf":
+                    exe_path = os.path.join(dirpath, filename)
+                    self.run("chmod 777 {}".format(exe_path))
+
+    def set_libraries_ids(self):
```

Comment:
What we really need to do at some point is extract parts of the [macOS packaging code](https://github.com/appleseedhq/appleseed/blob/master/scripts/appleseed.package.py#L567-L832) from `appleseed.package.py` (which is the most up-to-date) into a separate module and/or repo and reuse it for the Blender and Maya plugins instead of duplicating the logic.

Reply:
I agree, I noticed that when implementing this. It's not fun code to write, but I can take a look. It was actually harder getting this script to work then getting the plugin to compile :p

Reply:
I can imagine! It took me a long time to refine it and make it robust in `appleseed.package.py`, that's why I think it's worth reusing those parts and sharing the code.